src/kw_time_and_date.sh

```diff
@@ -94,5 +96,21 @@ function days_in_the_month()
   month_number=${month_number:-$(date +%m)}
   year=${year:-$(date +%Y)}
 
-  cal "$month_number" "$year" | awk 'NF {DAYS = $NF}; END {print DAYS}'
+  if [[ $month_number == 2 ]]; then
+    if ((year % 4 != 0)); then
+      days=28
+    elif ((year % 100 != 0)); then
+      days=29
+    elif ((year % 400 != 0)); then
+      days=28
+    else
+      days=29
+    fi
```

Comment:
Can we use `||` in the `if` conditions for the shared cases?

Reply:
Also, did you check if the current set of tests covers these new cases?

Reply:
Not so simple to concatenate the tests as the previous tests must fail in order, but I'll see what can be done.